### ISSUE TITLE
Changes in API? 

### ISSUE THREAD

**Issue by nadjaheitmann** [theforeman/foreman_puppet#150]
Puppet related Foreman Ansible Modules need to be updated after changes in the API. 

There might be need for more discussion about API changes in general.

**Comment by ezr-ondrej** [comment]
This should probably get to the 1.0 release, I'll align that.
If there is any volunteer who would want to take this effort, split it into more actionable tasks and be in charge of that, it's most definitelly welcome!

Thanks for reporting :+1:

**Comment by nadjaheitmann** [comment]
I don't think that there is a straight way forward to go here. So I put this up for discussion: 
https://community.theforeman.org/t/the-road-to-making-puppet-optional/17983/23

I highly appreciate your opinions @ezr-ondrej @timogoebel @bsuttner and everyone else.

**Comment by ezr-ondrej** [comment]
I agree, this can be closed as it mostly works, if something does not, we should open specific issues :)
